contrib_bots/lib/githubcomment.py

```diff
@@ -0,0 +1,148 @@
+
+
+class GitHubHandler(object):
+    '''
+    This plugin allows you to comment on a GitHub issue, under a certain repository.
+    It looks for messages starting with '@comment' or '@gcomment'.
+    '''
+
+    def usage(self):
+        return '''
+            This bot will allow users to comment on a GitHub issue.
+            Users should preface messages with '@comment' or '@gcomment'.
+            You will need to have a GitHub account.
+
+            Before running this, make sure to get a GitHub OAuth token.
+            The token will need to be authorized for the following scopes:
+            'gist, public_repo, user'.
+            Store it in the 'github_token.txt' file.
+            The 'github_token.txt' file should be located at '~/github_token.txt'.
+            Please input info like this:
+            '/<username>/<repository_owner>/<repository>/<issue_number>/<your_comment>'.
+            '''
+
+    def triage_message(self, message):
+        # return True iff we want to (possibly) response to this message
+
+        original_content = message['content']
+
+        is_comment = (original_content.startswith('@comment') or
+                      original_content.startswith('@gcomment'))
+
+        return is_comment
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        # this handles the message if its starts with @comment
+        if original_content.startswith('@comment'):
+            handle_input(client, original_content, original_sender)
+
+        # handle if message starts with @gcomment
+        elif original_content.startswith('@gcomment'):
+            handle_input(client, original_content, original_sender)
+
+handler_class = GitHubHandler
+
+
+def send_to_github(user, repo_owner, repo, issue, comment_body):
+    session = requests.session()
+
+    oauth_token = get_github_token()
+    session.auth = (user, oauth_token)
+    comment = {
+        'body': comment_body
+    }
+    r = session.post('https://api.github.com/repos/%s/%s/issues/%s/comments' % (repo_owner, repo, issue),
+                     json.dumps(comment))
+
+    return r.status_code
+
+
+def get_values_message(original_content, client, original_sender):
+    # gets rid of whitespace around the edges, so that they aren't a problem in the future
+    message_content = original_content.strip()
+
+    # splits message by '/' which will work if the information was entered correctly
+    message_content = message_content.split('/')
+
+    try:
+        # this will work if the information was entered correctly
+        user = message_content[1]
+        repo_owner = message_content[2]
+        repo = message_content[3]
+        issue = message_content[4]
+        comment_body = message_content[5]
+
+        return {'user': user, 'repo_owner': repo_owner, 'repo': repo, 'issue': issue, 'comment_body': comment_body}
```

Comment:
A different way to style this is dict(user=user, repo_owner=repo_owner, repo=repo, ...).

I think the squiggly form makes sense when the keys are arbitrary things like people's names, etc., but when you're just essentially just returning a bunch of fields, I like the `dict(foo=foo, bar=bar)` scheme.